modules/sku/src/ProductManager.php

```diff
@@ -523,38 +518,55 @@ public function synchronizeStockData(array $stock = [], $storeId = '') {
 
       if (empty($langcode)) {
         // It could be for a different store/website, don't do anything.
-        return (new ResourceResponse($response));
+        return ($response);
       }
     }
 
-    $lock_key = 'synchronizeProduct' . $stock['sku'];
-
-    // Acquire lock to ensure parallel processes are executed one by one.
-    do {
-      $lock_acquired = $lock->acquire($lock_key);
-    } while (!$lock_acquired);
+    // Work with single and array:
+    if (array_key_exists("sku", $stock)) {
+      $stockArray = [$stock];
+    }
+    else {
+      $stockArray = $stock;
+    }
 
-    /** @var \Drupal\acm_sku\Entity\SKU $sku */
-    if ($sku = SKU::loadFromSku($stock['sku'], $langcode)) {
-      $this->logger->info('Updating stock for SKU @sku.', ['@sku' => $stock['sku']]);
+    foreach ($stockArray as $stock) {
 
-      if (isset($stock['is_in_stock']) && empty($stock['is_in_stock'])) {
-        $stock['quantity'] = 0;
+      if (!isset($stock['sku']) || !strlen($stock['sku'])) {
+        $this->logger->error('Invalid or empty product SKU.');
+        return ($response);
       }
 
-      $quantity = isset($stock['quantity']) ? $stock['quantity'] : 0;
+      // Acquire lock to ensure parallel processes are executed one by one.
+      // TODO @Malachy: Should we change the key and move outside loop?
```

Comment:
Please resolve if needed and remove TODO comment.

Reply:
I'd like to leave the TODO in place. I don't know the answer and this seems like a place where efficiencies can be made in the future so I want to flag it as such.